common/djangoapps/config_models/admin.py

```diff
@@ -78,3 +88,108 @@ def revert(self, request, queryset):
                 args=(target.id,),
             )
         )
+
+
+class ShowHistoryFilter(ListFilter):
+    """
+    Admin change view filter to show only the most recent (i.e. the "current") row for each
+    unique key value.
+    """
+    title = 'Status'
+    parameter_name = 'show_history'
+
+    def __init__(self, request, params, model, model_admin):
+        super(ShowHistoryFilter, self).__init__(request, params, model, model_admin)
+        if self.parameter_name in params:
+            value = params.pop(self.parameter_name)
+            self.used_parameters[self.parameter_name] = value
+
+    def has_output(self):
+        """ Should this filter be shown? """
+        return True
+
+    def choices(self, cl):
+        """ Returns choices ready to be output in the template. """
+        show_all = self.used_parameters.get(self.parameter_name) == "1"
+        return (
+            {
+                'display': 'Current Configuration',
+                'selected': not show_all,
+                'query_string': cl.get_query_string({}, [self.parameter_name]),
+            },
+            {
+                'display': 'All (Show History)',
+                'selected': show_all,
+                'query_string': cl.get_query_string({self.parameter_name: "1"}, []),
+            }
+        )
+
+    def queryset(self, request, queryset):
+        """ Filter the queryset. No-op since it's done by KeyedConfigurationModelAdmin """
+        return queryset
+
+    def expected_parameters(self):
+        """ List the query string params used by this filter """
+        return [self.parameter_name]
+
+
+class KeyedConfigurationModelAdmin(ConfigurationModelAdmin):
+    """
+    :class:`~django.contrib.admin.ModelAdmin` for :class:`.ConfigurationModel` subclasses that
+    use extra keys (i.e. they have KEY_FIELDS set).
+    """
+    date_hierarchy = None
+    list_filter = (ShowHistoryFilter, )
+
+    def queryset(self, request):
+        """
+        Annote the queryset with an 'is_active' property that's true iff that row is the most
+        recently added row for that particular set of KEY_FIELDS values.
+        Filter the queryset to show only is_active rows by default.
+        """
```

Comment:
I feel that this should be a feature of the manager of the ConfigurationModel class, since it's common to be interested in only the current values and to perform queries on them.